blog/content/post/2020-07-21-secure-app-coding-practices.md

```diff
@@ -0,0 +1,74 @@
+---
+title: Secure Application – Best coding practices
+abstract: 'When building an application, we must assume that it will be exposed to hackers at all times, but also to abuse by ordinary recipients. The danger of the first group seems obvious, but what kind of risk are other users?'
+cover: /covers/secure-app.jpg
+author: malwina.mika
+layout: post
+published: true
+date: 2020-07-22
+archives: "2020"
+
+tags:
+  - security
+  - snyk
+  - helmet
+  - hackers
+  - vulnerability
+categories:
+  - Development
+  - App Dev
+  - Security
+---
+
+When building an application, we must assume that it will be exposed to hackers at all times, but also to abuse by ordinary recipients. The danger of the first group seems obvious, but what kind of risk are other users? You must know that attempts to "cheat the system" are quite common, especially in the e-commerce industry. There is a group of users who count, for example, on errors in calculating discounts or the number of elements. You can expect that if you have an online store, there will be at least a few people who will try to use your site in such a way as to find a vulnerability. Therefore, we must be really careful when building an application. We cannot count on everyone to act according to our plan. A secure application is one in which the author assumes the opposite scenario - such that everyone wants to find a vulnerability in our system. Only such an approach, and then appropriate protection, can give us at least a substitute for security.
+
+We will try to approach the topic from a slightly easier side - what are the best practices for writing web applications?
```

Comment:
> what are the best practices for writing web applications?

we can add 'secure' to this sentence "_what are the best practices for writing **secure** web applications?_"

Reply:
Fixed

Reply:
what kind of risk bring standard users?